### DISCUSSION TITLE
6월 28일 고도화 주간 회의

### DISCUSSION THREAD

**Discussion by kang-hyungu** [woowacourse/prolog#1352] [General]
- 프론트, 백엔드 각각 개선하고 싶은 내용 정리하기
  - 레벨2 초반에 개선하고 싶다고 정리한 내용 확인하고
  - 레벨3에 실제로 개선하고 싶은 내용 정리하기
- 개선할 내용은 자신이 작업하는데 걸리는 시간이 대략 5시간 걸리는 분량으로 가져오기
- 위키에 정리해서 코치에게 확인 받기
  - 우선순위 및 작업 분량 검토
- 페어 진행 여부는 개선하고 싶은 주제에 따라 정할 예정

---

개선하려는 이유
개선 후 기대하는 내용
개선 사항이 어떻게 나아졌는지 알 수 있을까요?(측정 방법)

**Comment by kang-hyungu** [comment]
- 다음주 월요일(7/3)에 각자 개선하고 싶은 내용 정리해서 공유하기

**Reply by kang-hyungu** [reply]
레벨2에서 작성한 개선하고 싶은 아이템 
#1231

**Reply by BGuga** [reply]
도메인 분석 때 발견한 문제점 정리

https://github.com/woowacourse/prolog/issues/1340

**Comment by D0Dam** [comment]
# FE 개발 내용 정리 

### Js ts 마이그레이션

**-  기존 컴포넌트의 props에 타입 부여하기 (컴포넌트 2개씩 분담해서 하기)**
-  Api에 대한 명세를 찾고 명세를 타입으로 정리하기, 정리한 타입을 모든 api요청에 적용하기
-  기존 hook들에 제네릭 적용가능한지 확인한 후 더 범용적이게 hook을 확장하기
-  이외 각 파일들에 들어간 이벤트 관련 타입 정의 및 추론된 타입 좁혀주기
-  전역 상태 타입 정의하기
-  양이 너무 많아서 나눠야 한다.

**Reply by kang-hyungu** [reply]
- 개선하려는 이유
- 개선 후 기대하는 내용
- 개선 사항이 어떻게 나아졌는지 알 수 있을까요?(측정 방법)

**Reply by feb-dain** [reply]
- components
  - ~~Card(3)~~
  - ~~CreatableSelectBox(1)~~
  - ~~DropdownMenu(3)~~
  - ~~FilterList(3)~~
  - ~~GithubLogin(1)~~
  - ~~Introduction(3)~~
  - ~~Modal(2)~~
  - ~~NavBar(3)~~
  - ~~NotFound(2)~~
  - Pagination(2)
  - ~~ProfilePageSideBar(2)~~
  - ~~SearchBar(2)~~
  - ~~SelectBox(3)~~
  - ~~Tag(2)~~

- hooks
  - ~~useSnackBar~~

- contexts
- utils

**Reply by D0Dam** [reply]
### 개선하려는 이유

- 서비스가 발전하면서 API가 많아질 수록 나의 코드를 다른 개발자가 작업해야하는 경우가 생긴다.
- API내 여러 함수가 순차적으로 실행될 때, 이 함수의 인자는 어디서 오는거고 어떤 형식을 띄어야하는것인지 알아내기 위해 콘솔로그를 찍어보곤 했다.
- 기존 프롤로그의 코드를 읽고 이해하는데에 시간이 정말 많이 걸렸다.
- 단순 css 를 바꾸는 것 자체도 css in js 를 사용하였기 때문에 타입을 알 필요가 있었다.
- 타입스크립트는 컴파일러가 검사를 먼저 해주기 때문에 실행 전 버그를 찾을 수 있는 것도 장점이다. 

### 개선 후 기대하는 내용

- 통일성을 줄 수 있다.
- 프롤로그를 추후 작업하게 될 개발자들의 편의성을 고려한다.
- 타입스크립트의 경우 변수에 마우스만 올리면 어떤 형식을 갖고있는지 알 수 있다.
- 코드 실행 전에 ts 컴파일러 를 실행할 수 있어서 빌드하기 전에 오류를 알 수 있다.
    - 자바스크립트는 동적타이핑이라 컴파일러에서는 에러가 잡히지 않는다. 따라서 런타임 환경에서 에러가 발생한다.

### 개선 사항이 어떻게 나아졌는지 알 수 있을까요?(측정 방법)
- 통일성
  - 모든 파일을 ts와 tsx 파일로 통일할 수 있다.
  - js 파일의 예외를 생각하지 않아도 되어서 편안하다.
- 프롤로그를 추후 작업하게 될 개발자들의 편의성을 고려한다.
  - 컴포넌트가 전달받는 props 의 형식을 바로 파악해서 이 파악하는데 드는 비용을 줄일 수 있었다.
    - 예시로 프로필 베지의 props를 알기 위해서 저번 이슈에서 1 시간 이상을 소요했다.
    - 이 시간을 5분 내로 대폭 단축시킬 수 있다.
- 타입스크립트의 경우 변수에 마우스만 올리면 어떤 형식을 갖고있는지 알 수 있다.
  - 이에 대해선 ts를 썻을 때 ide 의 환경에서 제공해주는 것이다.
  - 사소하다고 생각할 수 있지만, 해당 상수나 변수의 값을 찾아다니지 않고 마우스 호버로 바로 알 수 있어서 정말 편하다.

**Comment by mcodnjs** [comment]
- 인기있는 학습로그 알고리즘 수정
- 5기 학습로그 검색 (ES)
- GroupMember, MemberGroup 로직 수정
- LoginInterceptor의 CORS 로직 수정
  - https://github.com/woowacourse/prolog/issues/1325
- 모든 크루들에게 Badge 적용
  - 현재 4기 크루만 적용됨
  - https://github.com/woowacourse/prolog/issues/1324
- 학습로그 필터링에서 세션, 미션, 멤버 정렬 순서 수정

  - 최신순
      - findAllWithMySessionFirst()
      - findAllWithMyMissionFirst()
  - 멤버의 경우 본인 이름이 가장 상단에?

- BadRequest에 CustomException 적용

**Comment by wonyongChoi05** [comment]
5시간은 아니지만 일단 생각해봤슴다
* ES
* * 5기부터 동작하지 않는다.
* updateMember 이미지 변경 문제
* * 이미지 변경이 되지 않는다.(URL 문제인듯)
* jwt 관련 코드 리팩터링
* * jwt를 쓰는건지 안쓰는건지 모르곘음(특히 리프레시 토큰의 존재 유무)
* * jwt 안에서 정보를 뽑는 argumentresolver가 존재하지만,
* * 토큰 자체를 검증하는 필터가 필요해보임(토큰 자체가 prolog에서 발행한 토큰인지도 알아야하는 등등..)
* 예외처리 코드 리팩터링
* 커스텀 예외 처리가 알 수 없는 흐름으로 동작함(커스텀 예외로 등록을 안한다면 NotFoundErrorCodeException 발생
* 로그인 문제
* * 로그인이 너무 자주 풀림

**Comment by This2sho** [comment]
1. 에쎄이-답변의 수정 기능을 구현한다.
- 개선하려는 이유: 현재 애쎄이 답변 작성은 가능하지만 수정하는 기능이 존재하지 않음.
- 개선 후 기대하는 내용: 애쎄이 답변을 작성하고 내용을 수정할 수 있게 구현한다.
- 개선 사항이 어떻게 나아졌는지 알 수 있을까요?: 작성한 애쎄이 답변 페이지에 **수정 버튼(프론트)** 추가 및 클릭시 기존 내용을 수정할 수 있게

2. #1199
개선하려는 이유: 인기있는 학습로그 코드가 너무 복잡함.
개선 후 기대하는 내용: 우테코 컨벤션을 적용해서 깔끔하게 만든다.
측정 방법: 정양적 기준 최대 10줄을 적용하여 측정한다.

**Comment by BGuga** [comment]
1. 프로필 이미지 변경 기능

package wooteco.prolog.member.application;
MemberService.updateMember();

현재 방식은

수정 버튼을 누르면 

ProfileStudylogController
를 통해서 

<img width="1400" alt="image" src="https://github.com/woowacourse/prolog/assets/103228463/df2dcda9-d364-4574-8579-3f25e1d8cba3">



<img width="1434" alt="Pasted Graphic" src="https://github.com/woowacourse/prolog/assets/103228463/a5e0c2be-9615-424e-b1ec-d462e7b2c3b4">


이렇게 바꿀 수 있음

근데 imageURL 255자가 넘으면 에러 발생 -> image 주소만을 저장하는데 보통 이미지 주소는 255자로 정의되지 않음

S3 구현 잔재가 있는데(https://github.com/woowacourse/prolog/blob/main/backend/src/main/java/wooteco/prolog/image/application/ImageService.java) 이 부분 파악 후 사용하는 방향으로 진행할 수 있을 것 같음
원래 해당 기능은 학습로그에 이미지를 추가하기 위한 기능이었다고 함.( 슬로라는 전 기수가 작업한 내용 )
우선적으로 이미지 변경을 구현한 후 학습로그에 이미지 추가 기능 구현까지 확장할 수 있음
다만 권한 문제로 인해서 흐지부지 되었다고 전해들음

2. 뱃지 수여 기능 구현

- DB 접근해서 Id 확인 과정 필요

4기는 부여 받았음
https://prolog.techcourse.co.kr/usageness

뱃지 부여 로직을 구현하면 프론트에서 해당 이미지를 랜더링 하는 방식으로 구현되어있음
https://github.com/woowacourse/prolog/blob/main/frontend/src/assets/images/badge-2-level2.svg

뱃지 이미지 추가 제작 + 뱃지 부여 로직 구현시 뱃지 수여 가능

다만 현재 방식은 DB 세션 아이디를 직접 봐야하기 때문에 DB 접근 권한을 받아야함

그 외 내용 - bugfix

다른 사람 프로필 보다가 내 프로필 보면 기존에 보던 프로필이 출력된다

**Comment by kang-hyungu** [comment]
- 한 페어(베베, 호이)
  - [ ] 커스텀 예외 구조 잡기
    - 예외처리 코드 리팩터링
    - 커스텀 예외 처리가 알 수 없는 흐름으로 동작함(커스텀 예외로 등록을 안한다면 NotFoundErrorCodeException 발생
- 어나더 페어(라온, 푸우)
  - [ ] 인기있는 학습로그 조회 기능을 리팩터링한다.
    - [ ] LoginInterceptor의 CORS 로직 수정

---

- 개선하려는 이유
- 개선 후 기대하는 내용
- 개선 사항이 어떻게 나아졌는지 알 수 있을까요?(측정 방법)
